### ISSUE TITLE
Should we apply face alignment for evaluation data?

### ISSUE THREAD

**Issue by lymanblue** [wujiyang/Face_Pytorch#6]
Hi~

Thank you for your great work.

Does the reported accuracy result on validation data (e.g. LFW, MegaFace) apply face alignment process (e.g., MTCNN)?

Thank you.

**Comment by wujiyang** [comment]
@lymanblue  All the training and validation/test data have been aligned by MTCNN to the size of 112*112

**Comment by lymanblue** [comment]
Thanks.

And there is no file like ms1m.py in dataset directory.
Would you provide the file in the future?
Or is the data loader of ms1m the same with other dataset loader?

Does the training results of the cleaned-MS1M is the same with the MS1M-V2 provided from InsightFace?

Thank you.

**Comment by wujiyang** [comment]
@lymanblue  The cleaned-MS1M I used is provided by DeepGlint, it only has 3.9M images, while InsightFace has a 5.8M cleaned version.  
 The dataloader of MS1M is as same as the CASIA-WebFace.

**Comment by lymanblue** [comment]
Therefore, we have to preprocess (e.g., face alignment) the cleaned-MS1M of DeepGlint by ourself with the aid of the msra_lmk file.

On the other hand, ff we use the MS1M-V2 (already aligned?) from InsightFace.
Can we use the CASIA-WebFace loader directly for training?

Thank you.

**Comment by lymanblue** [comment]
Is the MS1M-IBUG from InsightFace the cropped and aligned result of the cleaned-MS1M?

**Comment by wujiyang** [comment]
You can use your own data (MS1M-V2) to train the models directly.

**Comment by lymanblue** [comment]
For training the model directly from MS1M-V2 from InsightFace.

Do you mean the following steps? (e.g., LFW for validation)
- set --train_root to the faces_emore/train.rec in train.py
- set --train_file_list to the faces_emore/train.idx in train.py
- set --lfw_test to the faces_emore/lfw.bin in train.py
- set --lfw_file_list to the path of txt file (i.e., from http://vis-www.cs.umass.edu/lfw/pairs.txt)

Thank you.

**Comment by wujiyang** [comment]
No, train.rec and train.idx are the mxnet's data format, the dataloader I provided is suitable for images,just like this:
00000  
    00000-00001.jpg
    00000-00002.jpg
    00000-00003.jpg
00001  
    00001-00001.jpg
    00001-00002.jpg
    00001-00003.jpg

**Comment by lymanblue** [comment]
Thank you~!

Could we use the the prepare_data.py from https://github.com/TreB1eN/InsightFace_Pytorch to convert the mxnet's format to the specified format? 
The data format looks similar. (if identical would be better).

**Comment by wujiyang** [comment]
Well, you can use it to parse the train.rec file to get original images.

**Comment by ypw-lbj** [comment]
model

**Comment by ypw-lbj** [comment]
请问您使用预训练模型了吗？
Have you used the pre training model?
